src/cool_seq_tool/sources/uta_database.py

```diff
@@ -954,3 +954,28 @@ def schema(self) -> str | None:
         """Create schema property."""
         path_elems = self.path.split("/")
         return path_elems[2] if len(path_elems) > 2 else None
+
+    @property
+    def sanitized_url(self) -> str:
+        """Sanitized DB URL with the password masked"""
+        netloc = ""
+        if self.username:
+            netloc += self.username
+            if self.password is not None and self.password != "":
+                netloc += ":***"
+            netloc += "@"
+        if self.hostname:
+            netloc += f"{self.hostname}"
+        if self.port:
+            netloc += f":{self.port}"
+
+        return urlunparse(
+            (
+                self.scheme,
+                netloc,
+                self.path,
+                self.params,
+                self.query,
+                self.fragment,
+            )
```

Comment:
I would really prefer we use a named tuple here, cause it's not obvious what order these should be in.

```suggestion
            UrlLibParseResult(
                scheme=self.scheme,
                netloc=netloc,
                path=self.path,
                params=self.params,
                query=self.query,
                fragment=self.fragment,
            )
```